File: db/conference_link.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in conference_link.
*/

class Conference_Link extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "conference_link";
    $this->order = "";
    $this->domain = "conference";
    $this->field['conference_link_id']['type'] = 'SERIAL';
    $this->field['conference_link_id']['not_null'] = true;
    $this->field['conference_id']['type'] = 'INTEGER';
    $this->field['conference_id']['not_null'] = true;
    $this->field['link_type_id']['type'] = 'INTEGER';
    $this->field['link_type_id']['not_null'] = true;
    $this->field['url']['type'] = 'VARCHAR';
    $this->field['url']['length'] = 1024;
    $this->field['url']['not_null'] = true;
    $this->field['title']['type'] = 'VARCHAR';
    $this->field['title']['length'] = 128;
    $this->field['description']['type'] = 'VARCHAR';
    $this->field['description']['length'] = 128;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['conference_link_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/conference_link.rb
module Momomoto
  class Conference_link < Base
    def initialize
      super
      @domain = 'conference'
      @fields = {
        :conference_link_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :conference_id => Datatype::Integer.new( {:not_null=>true} ),
        :url => Datatype::Varchar.new( {:not_null=>true, :length=>1024} ),
        :description => Datatype::Varchar.new( {:length=>256} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
